r2-core/src/main/java/com/linkedin/r2/message/timing/TimingKey.java

```diff
@@ -32,6 +35,7 @@
 public class TimingKey
 {
   private static final Map<String, TimingKey> _pool = new ConcurrentHashMap<>();
+  private static final ExecutorService _unregisterExecutor = Executors.newFixedThreadPool(1);
```

Comment:
Instead of creating a new thread pool at the `TimingKey` level, can we pass the executor to the `unregisterKey` function and leverage the standard executor passed around?

One option is to pass down the standard executor `_shutdownCallbackExecutor` from `HttpClientFactory`

Reply:
`_shutdownCallbackExecutor` is a boolean in HttpClientFactory.
TimingKey may be used more wildly than in `HttpClientFactory`, so that a separate thread provides better usability for other clients as well.
Any specific concerns against a new thread?

Reply:
If there are 100s of clients in service, that will end up creating 100s of threads. Having the executor run this make sure there is an upper bound on the total number of threads created.

Also, it is not applicable in this case - it is always encouraged to use executor so that other SI infra works as expected like IC context

Reply:
Hmm, the ConcurrentHashMap is static and shared between clients, so that I made the executor static and shared between clients.